js/app.js

```diff
@@ -67,55 +134,117 @@ function createInitialProducts() {
     fragment.appendChild(imgElement);
   }
   container.appendChild(fragment);
-  renderProducts();
-  // Add event listeners for each product image
-  for (let img of document.getElementsByClassName('productImage')) {
-    img.addEventListener('click', handleProductClick);
+  // If there are no current products to be displayed from a previous load, i.e., the currentProductArrayIndices variable is empty, then create initial products to display, otherwise render the products from the previous page load.
+  if (currentProductArrayIndices.length === 0) {
+    renderProducts();
+  } else {
+    setCurrentProducts();
   }
 }
 
 // This function renders random images on the page
 function renderProducts() {
-  // Define empty hash map for ensuring images rendered are not duplicates
-  let newProductArrayIndices = {};
-  for (let img of document.getElementsByClassName('productImage')) {
-    let index = getRandomIndex();
-
+  // Define empty array for ensuring images rendered are not duplicates
+  let newProductArrayIndices = [];
+  const imgArray = document.getElementsByClassName('productImage');
+  for (let i = 0; i < NUM_OF_PRODUCTS; i++) {
+    let randomIndex = getRandomIndex();
```

Comment:
Glad to see you decided to refactor this! Arrays have more inherent methods that are useful when you're working with a list of things than objects do, so this was a great change!